File: SIN BDD/RESTFULL_DOTNET_CONUNI_GR06/RESTFULL_DOTNET_CONUNI_GR06/05. CLIESCRITORIO/cliente_escritorio_rest_dotnet_conuni_gr06/src/Servicio/ServicioAutenticacion.cs
using Ec.Edu.Monster.Modelo;
using Ec.Edu.Monster.Utils;

namespace Ec.Edu.Monster.Servicio;

public class ServicioAutenticacion
{
    private readonly ClienteRest cliente = new();

    public Resultado Autenticar(string usuario, string contrasena)
    {
        return cliente.IniciarSesion(usuario, contrasena);
    }
}

File: SIN BDD/RESTFULL_JAVA_CONUNI_GR06/06. CLIWEB/cliente_web_restful_java_conuni_gr06/src/java/ec/edu/monster/modelo/ServicioAutenticacion.java
package ec.edu.monster.modelo;

public class ServicioAutenticacion {

    private final ClienteRest clienteRest = new ClienteRest();

    public boolean iniciarSesion(String usuario, String contrasena) throws Exception {
        String cuerpo = "{\"usuario\":\"" + escapar(usuario) + "\",\"contrasena\":\"" + escapar(contrasena) + "\"}";
        String respuesta = clienteRest.post("/sesion", cuerpo);
        return ClienteRest.extraerBooleano(respuesta, "valido");
    }

    private static String escapar(String texto) {
        if (texto == null) return "";
        return texto.replace("\\", "\\\\").replace("\"", "\\\"");
    }
}
